### PR TITLE
Update result.ts documentation

### LINKED ISSUE #128: result documentation

### PR DESCRIPTION
**Pull request by Elli610**
closes #128

### CODE REVIEW

File: packages/as-types/assembly/result.ts [hunk 1]
```diff
@@ -55,7 +70,9 @@ export class Result<T> {
   }
 
   /**
-   * @returns the value
+   * Get the value of the Result
+   *
+   * @returns the value of the Result
```
Comment (windushka) [inline packages/as-types/assembly/result.ts:75]:
```suggestion
   * @returns the value of the Result
   * @remarks This method retrieves the value of the Result and also handles nullable types.
```

**Review by windushka — APPROVED** [review]

File: packages/as-types/assembly/result.ts [hunk 2]
```diff
@@ -6,32 +6,43 @@
...
    *
    * @param value - expected value for passing case
    * @param error - error message for non-passing case
+   *
    */
   constructor(private value: T, public error: string | null = null) {}
 
   /**
-   * Checks that the result is okay.
+   * Checks if the Result is okay or not.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:18]:
```suggestion
   * Determines if the `Result` represents a successful outcome.
```

File: packages/as-types/assembly/result.ts [hunk 3]
```diff
@@ -6,32 +6,43 @@
...
+   *
    */
   @inline
   isOk(): bool {
     return !this.error;
   }
 
   /**
-   * Checks that the result is okay.
+   * Checks if the Result is okay or not.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:29]:
```suggestion
   * Determines if the `Result` represents a failed outcome.
```

File: packages/as-types/assembly/result.ts [hunk 4]
```diff
@@ -6,32 +6,43 @@
...
+   *
    */
   @inline
   isErr(): bool {
     return !this.isOk();
   }
 
   /**
-   * Checks that the result is okay and panic if not.
+   * Returns the value if the Result is okay. If the Result is not okay, it throws an assertion
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:40]:
```suggestion
   * Returns the value if the `Result` is successful. Throws an assertion error if not.
```

File: packages/as-types/assembly/result.ts [hunk 5]
```diff
@@ -6,32 +6,43 @@
...
    *
-   * @param msg - panic message that will prefix the error content.
+   * @param msg - A string representing the message to be shown on error.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:42]:
```suggestion
   * @param msg - The message to be displayed if the `Result` has failed.
```

File: packages/as-types/assembly/result.ts [hunk 6]
```diff
@@ -6,32 +6,43 @@
...
+   * @returns
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:43]:
```suggestion
   * @returns The value if the `Result` is successful.
```

File: packages/as-types/assembly/result.ts [hunk 7]
```diff
@@ -43,7 +54,11 @@ export class Result<T> {
   }
 
   /**
-   * Get the value. Panic if error.
+   * Get the value of the Result. Panic if error.
+   *
+   * @returns
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:59]:
```suggestion
   * @returns The value of the `Result` if successful.
```

File: packages/as-types/assembly/result.ts [hunk 8]
```diff
@@ -43,7 +54,11 @@ export class Result<T> {
   }
 
   /**
-   * Get the value. Panic if error.
+   * Get the value of the Result. Panic if error.
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:57]: Panic if error. should be in remarks

**Review by Ben-Rey — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/result.ts [hunk 9]
```diff
@@ -6,32 +6,42 @@
  */
```
Comment (Ben-Rey) [inline packages/as-types/assembly/result.ts:6]: Can you improve the top documentation by taking inspiration from massa-as-sdk Address class

**Review by Ben-Rey — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/result.ts (hunk 9 above)
Comment (Elli610) [reply packages/as-types/assembly/result.ts:6]: yes

**Review by Ben-Rey — APPROVED** [review]

**Review by gregLibert — APPROVED** [review]

**Pull request merged**